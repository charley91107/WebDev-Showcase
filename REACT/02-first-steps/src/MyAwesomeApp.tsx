import type { CSSProperties } from 'react';

export function MyAwesomeApp() {
    const lastName = "Mu√±oz";
    const firstName = " Carlos";

    const favoriteGames = [
        "The Legend of Zelda",
        "Super Mario",
        "Pokemon"
    ];
    const isActive = false;

    const address = {
        zipCode: 'ABC-123',
        country : 'Canada'
    };

    const myStyles: CSSProperties = {
        backgroundColor: '#fafafa',
        borderRadius: isActive ? 10 : 0,
        padding: 10
    }

    return(
        <>
            <h1 data-testid ="first-name-title">{firstName}</h1>
            <h3>{ lastName }</h3>

            <p>{ favoriteGames.join(", ") }</p>
            <p>{2 + 2}</p>

            <h1>{ isActive ? "Active" : "Inactive"}</h1>
            <p style={myStyles}>{JSON.stringify(address)}</p>
        </>
    )
}