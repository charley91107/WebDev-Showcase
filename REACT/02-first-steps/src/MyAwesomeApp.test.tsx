import { render, screen } from '@testing-library/react';
import { describe, expect, test } from "vitest";

import { MyAwesomeApp } from './MyAwesomeApp';


describe('MyAwesomeApp', () => {
    test('should render firstName and lastName', () => {
        const { container } = render (<MyAwesomeApp />);
        //screen.debug();

        const h1 = container.querySelector('h1');
        const h3 = container.querySelector('h3');
        
        expect(h1?.innerHTML).toContain('Carlos');
        expect(h3?.innerHTML).toContain('MuÃ±oz');
    });

    test('should render firstName and lastName - screen', () => {
        render (<MyAwesomeApp />);
        screen.debug();

        const h1 = screen.getByTestId('first-name-title');
        expect(h1.innerHTML).toContain('Carlos');
    });

    test('should match snapshot', () => {
        const { container } = render(<MyAwesomeApp />)
        expect(container).toMatchSnapshot();
    })

    test('should match snapshot2', () => {
        render(<MyAwesomeApp />)
        expect( screen.getByTestId('div-app') ).toMatchSnapshot();
    }) 
})