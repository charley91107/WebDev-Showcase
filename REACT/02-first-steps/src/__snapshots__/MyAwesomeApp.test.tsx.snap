// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`MyAwesomeApp > should match snapshot 1`] = `
<div>
  <div
    data-testid="div-app"
  >
    <h1
      data-testid="first-name-title"
    >
      Carlos
    </h1>
    <h3>
      Muñoz
    </h3>
    <p
      class="mi-clase-favorita"
    >
      The Legend of Zelda, Super Mario, Pokemon
    </p>
    <p>
      4
    </p>
    <h1>
      Inactive
    </h1>
    <p
      style="background-color: rgb(250, 250, 250); border-radius: 0; padding: 10px;"
    >
      {"zipCode":"ABC-123","country":"Canada"}
    </p>
  </div>
</div>
`;

exports[`MyAwesomeApp > should match snapshot2 1`] = `
<div
  data-testid="div-app"
>
  <h1
    data-testid="first-name-title"
  >
    Carlos
  </h1>
  <h3>
    Muñoz
  </h3>
  <p
    class="mi-clase-favorita"
  >
    The Legend of Zelda, Super Mario, Pokemon
  </p>
  <p>
    4
  </p>
  <h1>
    Inactive
  </h1>
  <p
    style="background-color: rgb(250, 250, 250); border-radius: 0; padding: 10px;"
  >
    {"zipCode":"ABC-123","country":"Canada"}
  </p>
</div>
`;
